src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1012,6 +1013,20 @@ public static void onEmptyLeftClick(EntityPlayer player, ItemStack stack)
         MinecraftForge.EVENT_BUS.post(new PlayerInteractEvent.LeftClickEmpty(player, stack));
     }
 
+    public static CriticalHitEvent getCriticalHit(EntityPlayer player, Entity target, boolean vanillaCritical, float damageModifier)
+    {
+        CriticalHitEvent hitResult = new CriticalHitEvent(player, target, damageModifier);
+        MinecraftForge.EVENT_BUS.post(hitResult);
+        if (hitResult.getResult() != Event.Result.DENY)
```

Comment:
Since this piece of code is very tricky (I also had to read it a couple times to confirm it was right) can you write is this way instead?
```java
if (hitResult.getResult() == Event.Result.ALLOW ||
    (vanillaCritical && hitResult.getResult() == Event.Result.DEFAULT))
{
    return hitResult;
}
return null;
```